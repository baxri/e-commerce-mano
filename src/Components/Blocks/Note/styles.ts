import themeGet from '@styled-system/theme-get';
import styled, { css } from 'styled-components/native';
import { margin } from 'styled-system';

export const Container = styled.TouchableOpacity`
  padding: 16px;
  flex-grow: 1;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  border-radius: 16px;
  border-width: 1px;
  background-color: ${themeGet('common.colors.greyLightExtra')};
  border-color: ${themeGet('common.colors.altLight')};

  ${margin}
`;
