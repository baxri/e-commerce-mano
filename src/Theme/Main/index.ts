import button from "./button";
import colors from "./colors";
import common from "./common";
import fonts from "./fonts";

export default {
  name: "Main Theme",
  dark: false,
  button,
  common,
  colors,
  fonts,

  // --   0  1  2  3  4   5   6   7   8   9
  space: [0, 2, 4, 8, 12, 16, 20, 24, 32, 64],
  // --       0   1   2   3    4   5   7   8   9
  fontSizes: [10, 12, 14, 16, 18, 20, 22, 24, 26],
  // --         0   1   2   3    4   5   7   8   9
  lineHeights: [13, 15, 17, 19, 21, 23, 25, 27, 29],
  // --         0    1    2    3
  fontWeights: [400, 500, 600, 700],
};
