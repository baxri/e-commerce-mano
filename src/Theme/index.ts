import MainTheme from "./Main";

enum ThemeID {
  Main = "main",
}

const THEMES: Record<ThemeID, any> = {
  [ThemeID.Main]: MainTheme,
};

export { ThemeID, THEMES };
