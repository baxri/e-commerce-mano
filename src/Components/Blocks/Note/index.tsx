import React from "react";
// @ts-ignore
import { pick } from "@styled-system/props";
import { MarginProps } from "styled-system";

import { Column, Text } from "Components/UI";

import { Container } from "./styles";

type NoteProps = MarginProps & {
  text: string;
};

function Note({ text, ...rest }: NoteProps) {
  return (
    <Container {...pick(rest)}>
      <Column stretch>
        <Text color="grey" fontSize={1} ml={5}>
          {text}
        </Text>
      </Column>
    </Container>
  );
}

export default Note;
