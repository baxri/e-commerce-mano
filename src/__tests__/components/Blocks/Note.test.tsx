import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';

import { ThemeID, THEMES } from 'Theme';

import Note from 'Components/Blocks/Note';

test('Note', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={THEMES[ThemeID.Main]}>
        <Note text="Personal info" />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
