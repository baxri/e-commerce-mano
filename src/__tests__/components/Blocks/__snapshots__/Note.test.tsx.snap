// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FBFCFF",
      "borderColor": "#EFF5FF",
      "borderRadius": 16,
      "borderWidth": 1,
      "flexDirection": "row",
      "flexGrow": 1,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      [
        {
          "flexDirection": "column",
        },
      ]
    }
  />
  <View
    stretch={true}
    style={
      [
        {
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      color="grey"
      fontFamily="Inter"
      fontSize={1}
      fontWeight={0}
      ml={5}
      style={
        [
          {
            "color": "#6D758E",
            "fontFamily": "Inter",
            "fontSize": 12,
            "fontWeight": "400",
            "marginLeft": 16,
          },
        ]
      }
    >
      Personal info
    </Text>
  </View>
</View>
`;
